Tools/i18n/pygettext.py

```diff
@@ -279,6 +283,272 @@ def getFilesForName(name):
+        if not state.is_plural:
+            raise ValueError(f'{state.filename}:{state.lineno}: '
+                             'Missing msgid_plural section')
+        index = int(match.group(1))
+        line = line.removeprefix(match.group())
+        if state.msgstr is None:
+            state.msgstr = []
+        next_plural_index = len(state.msgstr)
+        if index != next_plural_index:
+            raise ValueError(f'{state.filename}:{state.lineno}: '
+                             'Plural form has incorrect index, found '
+                             f"'{index}' but should be '{next_plural_index}'")
+        state.msgstr.append(parse_quoted_strings(state, line))
+    else:
+        # This is a regular (non-plural) msgstr
+        if state.is_plural:
+            raise ValueError(f'{state.filename}:{state.lineno}: '
+                             'Indexed msgstr required after msgid_plural')
+        if state.section == POSection.STR:
+            raise ValueError(f'{state.filename}:{state.lineno}: '
+                             'msgstr not allowed after msgstr')
+        line = line.removeprefix('msgstr')
+        state.msgstr = parse_quoted_strings(state, line)
+    state.section = POSection.STR
+
+
+def parse_line(state, line):
+    line = parse_quoted_strings(state, line.decode(state.encoding))
+    if state.section == POSection.CTXT:
+        state.msgctxt += line
+    elif state.section == POSection.PLURAL:
+        state.msgid_plural += line
+    elif state.section == POSection.ID:
+        state.msgid += line
+    elif state.section == POSection.STR:
+        if isinstance(state.msgstr, list):
+            # This belongs to the last msgstr[N] entry
+            state.msgstr[-1] += line
+        else:
+            state.msgstr += line
+    else:
+        raise ValueError(f'{state.filename}:{state.lineno}: '
+                         f'Syntax error before:\n{line}')
+
+
+def parse_quoted_strings(state, line):
+    """
+    Parse a line containing one or more quoted PO strings separated
+    by whitespace.
+
+    Example: "Hello, " "world!" -> 'Hello, world!'
+    """
+    line = line.strip()
+    if not line:
+        return ''
+
+    quoted_string = r'"([^"\\]|\\.)*"'
+    # One or more quoted strings, possibly separated by whitespace
+    quoted_strings = fr'^({quoted_string}\s*)+$'
+
+    if not re.match(quoted_strings, line):
+        raise ValueError(f'{state.filename}:{state.lineno}: '
+                         f'Syntax error: {line}')
+
+    string = ''
+    for match in re.finditer(quoted_string, line):
+        part = match.group()
+        string += parse_quoted_string(state, part)
+    return string
+
+
+def parse_quoted_string(state, string):
+    """Parse a single quoted PO string."""
+    # Check if there are any disallowed escape sequences
+    # The allowed escape sequences are:
+    #   - \n, \r, \t, \\, \", \a, \b, \f, \v
+    #   - Octal escapes: \o, \oo, \ooo
+    #   - Hex escapes: \xh, \xhh, ...
```

Comment:
This is not standardized but I checked this with GNU `msgfmt` and these are the ones that are allowed.